### ISSUE TITLE
JSON Output for restic check command

### ISSUE THREAD

**Issue by baernm** [restic/restic#1378]
## Output of `restic version`

restic 0.7.3
compiled with go1.9 on linux/amd64

## How did you run restic exactly?
restic -r rest:path check --json

## What backend/server/service did you use?
rest-server

## Expected behavior

A JSON array containing information about the check results, at least something like "errors":0 oder "errors":1 would be sufficient for me

## Actual behavior

no json output given

## Steps to reproduce the behavior
Just use restic with check command and json flag

**Comment by fd0** [comment]
Thanks for the suggestion, what's your use case?

**Comment by baernm** [comment]
We're using a php script to run regular checks on our repos and this would help us using the json format consistently and simplify this task a lot.
So, we would appreciate this feature very much! Thanks for your time!

**Comment by fd0** [comment]
Ah, maybe I should've asked more specifically: Are you just interested in the result of the check, as in "has failed" or "has succeeded"? If that's the case, then you can already check that automatically by looking at the exit code: When `restic check` exits with 0, all is fine. Otherwise errors have been detected.

Or are you also interested in the details of the errors?

**Comment by baernm** [comment]
Yes, that one we already found out. It is ok for the moment but for a future release it would indeed be nice to have more specifics about the error in the json array result

**Comment by russelldavies** [comment]
There are so many uses cases for having the output in JSON, in particular scripts that read the output and decide what to do (as opposed to looking at return codes and/or parsing the textual output).

I think the best approach would be to use [Go templates](https://golang.org/pkg/text/template/) like many other projects use.

**Comment by MichaelEischer** [comment]
The output format should be structured similar to that of the backup command.

**Comment by darkdragon-001** [comment]
There is #4581 working on the implementation.
